### ISSUE TITLE
Email Extract Performance

### ISSUE THREAD

**Issue by ghost** [hedii/extractors#3]
I have been using this extractor for a while and recently have noticed that some sites really hamper performance. Originally I thought this was to do with curl and slow responses, but I used Blackfire to track the issue down to the preg_match_all within EmailExtractor->extract. It seemed like when the extractor was given a DOM that contained a base64 encoded string (e.g. PNG), the preg_match_all was taking MUCH longer than usual.

I messed around with the regex and I'm getting much better performance from:

preg_match_all('/[a-z0-9_\-\+]{1,256}+@[a-z0-9\-]{1,256}+\.([a-z]{2,3})(?:\.[a-z]{2})?/i', $dom, $matches);

Basically limiting the user and domain to max. 256 characters. A crawl that was taking over 1 hour is now taking about 1 minute. Not sure if you want to implement this, but thought I'd flag it because it took me ages to track down.

Thanks for the good work!

**Comment by hedii** [comment]
Improving the performance of the regex is always good.

But your regex does not allows all the characters from the current regex, and it looks too strict.

For example:
- "." (period) in string before @ (my.address@example.com)
- "_" (underscore) in string after @ (contact@the_example.com)
- subdomain after @ (contact@sub.example.com)
- long domain extensions (contact@example.photography) *(I need to fix my own regex for this one)*
- etc...

If you make a PR with a regex with limited length, and provide some unit tests, I would be happy to merge it.

**Comment by ghost** [comment]
I actually got the regex from this SO post:

https://stackoverflow.com/questions/8175644/why-does-this-regex-take-so-long-to-find-email-addresses-in-certain-files

I'll see if I can have a play with the Regex based on your comments.

Thanks :)
